src/App.tsx

```diff
@@ -1,20 +1,89 @@
-import React from 'react';
+import React, { createContext, useState, useEffect } from 'react';
+import { Route, Routes, useSearchParams } from 'react-router-dom';
+import { ProductPage } from './pages/ProductPage';
+import { CatalogPage } from './pages/CatalogPage';
+import { NotFoundPage } from './pages/NotFoundPage';
+import { CardPage } from './pages/CartPage';
+
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ISearchContext, ICartContext } from './models';
+
+import { getLocalStorage, setLocalStorage } from './utils/localStorage';
+
+import { getCartCount } from './utils/cart/getCartCount';
+import { getCartTotal } from './utils/cart/getCartTotal';
+import { addProductsCart } from './utils/cart/addProductsCart';
+import { removeProductCart } from './utils/cart/removeProductCart';
+import { updateProductCartCount } from './utils/cart/updateProductCartCount';
+import { isAddCart } from './utils/cart/isAddCart';
+import { getCartDiscountTotal } from './utils/cart/getCartDiscountTotal';
+import { getCartCountProduct } from './utils/cart/getCartCountProduct';
+import { getCartCountLimit } from './utils/cart/getCartCountLimit';
+
+export const SearchContext = createContext<Partial<ISearchContext>>({});
+export const CartContext = createContext<Partial<ICartContext>>({});
 
 function App() {
+  const [search] = useSearchParams();
+  const [searchValueInput, setSearchValueInput] = useState(
+    search.get('search') || '',
+  );
+
+  const [cartCount, setCartCount] = useState(getCartCount() || 0);
+  const [cartTotal, setCartTotal] = useState(getCartTotal() || 0);
```

Comment:
Do not use function call in useState initiation. It runs this function on every render. Better use lazy state initiation
```suggestion
  const [cartTotal, setCartTotal] = useState(() => getCartTotal() || 0);
```